### ISSUE TITLE
DSHMv3 does not allow setting LEDs states from config file

### ISSUE THREAD

**Issue by Kanuan** [nefarius/DsHidMini#260]
# Context

In DS3's output report, there are:

- 1 ) 1 byte that sets LED states
- 2 ) 4 groups of 5 bytes, each one for each LED, that sets custom effects the associated LED should execute
    - Effects are only executed if the related LED has been enabled via (1)

# Issue:

Currently, DSHMv3 allows settings (2), but not (1), so that means the LEDs are always off and no effects are shown

# Proposed solution:

1. Allow dshidmini to read from a property in the .json file called "LEDsState" that sets the corresponding LEDs states byte in the output report
2. Rename the "Enabled" properties in LEDsCustoms to "EffectsEnabled" to make things clearer.

![image](https://user-images.githubusercontent.com/24910442/197353041-30499f8a-8587-44fa-a7cc-5cc450777f53.png)

**Comment by nefarius** [comment]
@Kanuan FYI I renamed the suggested `LEDsEnabled` to `LEDFlags` and `Enabled` to `EnabledFlags`

**Comment by Kanuan** [comment]
I might be doing something wrong, but the LED effects don't seem to be loading as expected.

1. I've connected the controller
2. Settings were already in custom LED settings, driver authority mode
3. Changed which LEDs were enabled to cause a hot-reload
4. This is what was sent to the controller:

![image](https://user-images.githubusercontent.com/24910442/200217875-68d019f1-c800-4298-bf6a-ea7a2e02dd3d.png)

In orange 1E means all LEDs are on (1e -> 11110), which is as expected.

The other 4 groups of 5 bytes are the effect related bytes:

- 1st byte -> Duration
- 2nd byte -> IntervalDuration
- 3rd byte -> EnableEffects (0x10 -> ON // 0x00 -> off)
- 4th byte -> Interval... ON?
- 5th byte -> Interval... OFF?

Below are my [DsHidMini.json](https://github.com/ViGEm/DsHidMini/files/9947826/DsHidMini.txt) LED settings:

![image](https://user-images.githubusercontent.com/24910442/200217985-7e422b3b-de05-48e5-8c3e-e4c60431df17.png)

-  By my settings, the 2nd byte should've been 0x0F, and the 4th and 5th bytes that are 0xFF in the output report should've been 0x7F
- I know the values I've used should be working because that's what is being used in DS4 lightbar translation to have a blinking LED, as shown in the image below:

![image](https://user-images.githubusercontent.com/24910442/200218072-8ad81638-ac97-4a2e-ad98-be781b262dda.png)

**Comment by Kanuan** [comment]
Oh, and besides that, when "cold-loading" the settings in custom LED mode all LEDs remain off until a hot-reload occurs

**Comment by nefarius** [comment]
> The other 4 groups of 5 bytes are the effect related bytes:
> 
> * 1st byte -> Duration
> * 2nd byte -> IntervalDuration
> * 3rd byte -> EnableEffects (0x10 -> ON // 0x00 -> off)
> * 4th byte -> Interval... ON?
> * 5th byte -> Interval... OFF?

You got byte 4 and 5 mixed up btw.:

```c
typedef struct _DS_LED
{
	UCHAR Duration;

	UCHAR IntervalDuration;

	UCHAR EnabledFlags;

	UCHAR IntervalPortionOff;

	UCHAR IntervalPortionOn;

} DS_LED, * PDS_LED;
```

**Comment by Kanuan** [comment]
On my .json settings the order is correct, but I'm setting the same value on both of them though, so it should work.

**Comment by Kanuan** [comment]
Sorry. I took the screenshot and then noticed the authority parameter was missing. I re-did the tests with the parameter then and got the same results but forgot to update the screenshot

**Comment by nefarius** [comment]
Understood, will investigate on my end.

**Comment by Kanuan** [comment]
Fixed in #348
